### ISSUE TITLE
Trying to follow "your first asp.net 5 app on mac"

### ISSUE THREAD

**Issue by alberttwong** [aspnet/aspnet-docker#137]
So I'm following http://docs.asp.net/en/latest/tutorials/your-first-mac-aspnet.html and yeoman generated a asp.net sample app (http://docs.asp.net/en/latest/client-side/yeoman.html).  I can run it just fine when I execute "dnx web".   I also noticed that yeoman generated a dockerfile with it.   Now the docker file uses "Kestrel" and I just changed it to "web" and "docker build" and "docker run".   I can see the server running but somehow my localhost:8080 won't nat into the docker instance. 

```
bash-3.2$ docker run -t -p 8080:5000 myapp
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (2ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      PRAGMA foreign_keys=ON;
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (1ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT COUNT(*) FROM "sqlite_master" WHERE "name" = '__EFMigrationsHistory' AND "type" = 'table';
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      PRAGMA foreign_keys=ON;
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT COUNT(*) FROM "sqlite_master" WHERE "name" = '__EFMigrationsHistory' AND "type" = 'table';
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      PRAGMA foreign_keys=ON;
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT "MigrationId", "ProductVersion"
      FROM "__EFMigrationsHistory"
      ORDER BY "MigrationId";
info: Microsoft.Extensions.DependencyInjection.DataProtectionServices[0]
      User profile is available. Using '/root/.local/share/ASP.NET/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
Hosting environment: Production
Now listening on: http://localhost:5000
Application started. Press Ctrl+C to shut down.
^[[24;1R
bash-3.2$ docker run -t -d -p 8080:5000 myapp
ad452dc9fe5aa5b2da293f3edba172976320c830087a3054ba740f8f6a38da0b
bash-3.2$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
ad452dc9fe5a        myapp               "dnx -p project.json "   6 seconds ago       Up 5 seconds        0.0.0.0:8080->5000/tcp   hungry_feynman
bash-3.2$ docker inspect ad452dc9fe5a |grep IPAddress
        "IPAddress": "172.17.0.19",
        "SecondaryIPAddresses": null,
bash-3.2$ 
```

my dockerfile
```
bash-3.2$ vi Dockerfile 

FROM microsoft/aspnet:1.0.0-rc1-update1

RUN printf "deb http://ftp.us.debian.org/debian jessie main\n" >> /etc/apt/sources.list
RUN apt-get -qq update && apt-get install -qqy sqlite3 libsqlite3-dev && rm -rf /var/lib/apt/lists/*

COPY . /app
WORKDIR /app
RUN ["dnu", "restore"]

EXPOSE 5000/tcp
ENTRYPOINT ["dnx", "-p", "project.json", "web"]
```
#
Any ideas?

**Comment by charliesolomon** [comment]
Port 5000 is the container port that is exposed, right?  8080 is the host port it is mapped to.  Did you try localhost:8080?

**Comment by alberttwong** [comment]
Yup..  tried both ports.

**Comment by friism** [comment]
Did the sample work before you changed anything?

Anywho, you need to get kestrel to bind to 0.0.0.0 (not just localhost): `docker run -t -p 5000:5000 myapp --server.urls http://0.0.0.0:5000`

Also, depending on how you run Docker on your Mac, containers are not available on `localhost` (this is a linux'ism). If you're using Docker Machine, you can find the relevant IP address with `docker-machine ip default`

**Comment by alberttwong** [comment]
bash-3.2$ docker run -t -p 5000:5000 yourapp --server.urls http://0.0.0.0:5000
Error: Unable to load application or execute command 'kestrel'. Available commands: web, ef.

**Comment by alberttwong** [comment]
so I modified the dockerfile back to web.

bash-3.2$ docker run -t -p 5000:5000 yourapp --server.urls http://0.0.0.0:5000
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (2ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      PRAGMA foreign_keys=ON;
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT COUNT(*) FROM "sqlite_master" WHERE "name" = '__EFMigrationsHistory' AND "type" = 'table';
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      PRAGMA foreign_keys=ON;
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT COUNT(*) FROM "sqlite_master" WHERE "name" = '__EFMigrationsHistory' AND "type" = 'table';
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      PRAGMA foreign_keys=ON;
info: Microsoft.Data.Entity.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (0ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT "MigrationId", "ProductVersion"
      FROM "__EFMigrationsHistory"
      ORDER BY "MigrationId";
info: Microsoft.Extensions.DependencyInjection.DataProtectionServices[0]
      User profile is available. Using '/root/.local/share/ASP.NET/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
Hosting environment: Production
Now listening on: http://0.0.0.0:5000
Application started. Press Ctrl+C to shut down.
^[[24;1R

**Comment by alberttwong** [comment]
So I figured out a way to get this to work.

1. Find out the docker ip

docker is configured to use the default machine with IP 192.168.99.100

2. must run docker run with server.urls

docker run -t -p 8080:5000 yourapp --server.urls http://0.0.0.0:5000

3. then you can open browser to http://192.168.99.100:8080/

**Comment by friism** [comment]
Great, recommend closing this issues.

It might be good to add a Docker section to the docs: https://github.com/aspnet/Docs

**Comment by alberttwong** [comment]
Found out that there is a fix from the generator guys so you don't have to change the docs.  https://github.com/OmniSharp/generator-aspnet/issues/579#issuecomment-203587270

**Comment by friism** [comment]
Well, the IP stuff is still confusing

On Wednesday, March 30, 2016, Albert Wong <notifications@github.com> wrote:

> Closed #137 <https://github.com/aspnet/aspnet-docker/issues/137>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/aspnet/aspnet-docker/issues/137#event-608890876>
>


-- 
http://friism.com/
(650) 318-1051
Sapere aude
